src/olympia/lib/akismet/admin.py

```diff
@@ -1,10 +1,122 @@
 from django.contrib import admin
+from django.shortcuts import get_object_or_404
+from django.utils.html import format_html
+from django.utils.translation import ugettext, ugettext_lazy as _
 
 from .models import AkismetReport
+from .tasks import submit_to_akismet
 
 
+class ReportedStatusFilter(admin.SimpleListFilter):
+    title = 'Reported Status'
+    parameter_name = 'reported_status'
+    REPORTED_HAM = 'ham'
+    REPORTED_SPAM = 'spam'
+    UNREPORTED = 'unreported'
+    QUERYSET_LOOKUP = {
+        REPORTED_SPAM: {'reported': True,
+                        'result__in': [AkismetReport.DEFINITE_SPAM,
+                                       AkismetReport.MAYBE_SPAM]},
+        REPORTED_HAM: {'reported': True,
+                       'result': AkismetReport.HAM},
+        UNREPORTED: {'reported': False},
+    }
+    VALUES = {
+        REPORTED_SPAM: ugettext('Reported Spam'),
+        REPORTED_HAM: ugettext('Reported Ham'),
+        UNREPORTED: ugettext('Not Reported'),
+    }
+
+    def lookups(self, request, model_admin):
+        return self.VALUES.items()
+
+    def queryset(self, request, queryset):
+        filters = self.QUERYSET_LOOKUP.get(self.value(), {})
+        return queryset.filter(**filters)
```

Comment:
Let's just not call `.filter` if there are no filters applied. Although Django does seem to handle this nicely, there's simply no need to to through all this.